test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs

```diff
@@ -1,34 +1,150 @@
+
+            _module = Path.Combine(_tempDirectory.FullName, Path.GetFileName(module));
+        }
+
+        public void Dispose()
+        {
+            _tempDirectory.Delete(true);
+        }
+
         [Fact]
         public void TestInstrument()
         {
-            string module = typeof(InstrumenterTests).Assembly.Location;
-            string pdb = Path.Combine(Path.GetDirectoryName(module), Path.GetFileNameWithoutExtension(module) + ".pdb");
-            string identifier = Guid.NewGuid().ToString();
+            var instrumenter = new Instrumenter(_module, _identifier);
+            var result = instrumenter.Instrument();
 
-            var directory = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), identifier));
+            Assert.Equal(Path.GetFileNameWithoutExtension(_module), result.Module);
+            Assert.Equal(_module, result.ModulePath);
+        }
 
-            File.Copy(module, Path.Combine(directory.FullName, Path.GetFileName(module)), true);
-            File.Copy(pdb, Path.Combine(directory.FullName, Path.GetFileName(pdb)), true);
+        private List<string> GetMethodsThatMustBeIncludedToCoverage(Type classType, Type excludeAttributeType)
+        {
+            var methodsForIncludingToCodeCoverage = classType
+                .GetMethods()
+                .Where(e => e.DeclaringType.Name == classType.Name &&
+                            (!e.CustomAttributes.Any() ||
+                             e.CustomAttributes.Any(a => a.AttributeType.Name != excludeAttributeType.Name)))
+                .Select(e => e.GetMethodNameInMonoCecilFormat())
+                .ToList();
+            return methodsForIncludingToCodeCoverage;
+        }
 
-            module = Path.Combine(directory.FullName, Path.GetFileName(module));
+        private List<string> GetMethodsThatMustBeExcludedFromCoverage(Type classType, Type excludeAttributeType)
+        {
+            var methodsForExcludingFromCodeCoverage = classType
+                .GetMethods()
+                .Where(e => e.CustomAttributes.Any(a => a.AttributeType.Name == excludeAttributeType.Name))
+                .Select(e => e.GetMethodNameInMonoCecilFormat())
+                .ToList();
+            return methodsForExcludingFromCodeCoverage;
+        }
 
-            Instrumenter instrumenter = new Instrumenter(module, identifier);
-            var result = instrumenter.Instrument();
+        public static List<object[]> ClassesWithExcludingAttributes = new List<object[]>
+        {
+            new object[] {typeof(ClassWithExcludeFromCoverage)},
+            new object[] {typeof(ClassWithExcludeFromCodeCoverage)}
+        };
 
-            Assert.Equal(Path.GetFileNameWithoutExtension(module), result.Module);
-            Assert.Equal(module, result.ModulePath);
+        [Theory]
+        [MemberData(nameof(ClassesWithExcludingAttributes))]
+        public void Instrument__ClassWithExcludeFromCodeCoverageAttributeMustExcludedFromResult(
```

Comment:
Please use the standard C# Pascal case syntax on all your method names and prefix test methods with `Test`. E.g. `TestClassWithExcludeFromCodeCoverageAttributeMustExcludedFromResult`

Reply:
Can you provide link to this conventions? I prefer the strategy by which the name of the unit test reflects: 1) the module that is being tested (constructor, method, property); 2) the input parameters for the test (if it exists); 3) the expected result. I see no point in the name of the method to specify the word "Test"...

Reply:
Here's a handy link: https://msdn.microsoft.com/en-us/library/ms229043(v=vs.100).aspx#Anchor_1. I definitely welcome the descriptiveness of your test methods, I just need them to follow the casing convention to be in line with the style of the code in the repo, for consistency's sake. The same goes for the `Test` prefix.

Reply:
This link describes only common naming rules and contains no information about tests naming rules. All examples of tests naming that I saw in docs.microsoft.com does not correspond this rules ([link1](https://docs.microsoft.com/en-us/visualstudio/test/unit-test-basics), [link2](https://docs.microsoft.com/en-us/aspnet/web-api/overview/testing-and-debugging/unit-testing-with-aspnet-web-api)). I think that is bad idea to add `Test` prefix because it is redundant. To maintain a single style of naming tests, I can change the names of already existing tests - it is not hard for me)) We can use this guide of ASP.NET engineering: https://github.com/aspnet/Home/wiki/Engineering-guidelines. I think that in any case we must describe contributing rules so that in the future there will be fewer questions - are you agree? :)

Reply:
Feel free to update the name of existing tests. Thanks.

Reply:
@tonerdo hi! I'm going to do it in the next 2 days, thanks)